pkg/proxy/ipvs/proxier_test.go

```diff
@@ -3892,3 +3892,57 @@ func TestFilterCIDRs(t *testing.T) {
 		t.Errorf("cidrs %v is not expected %v", cidrs, expected)
 	}
 }
+
+func TestLoadBalancerIngressRouteTypeProxy(t *testing.T) {
+	_, fp := buildFakeProxier()
+	svcIP := "10.20.30.41"
+	svcPort := 80
+	svcNodePort := 3001
+	svcLBIP := "1.2.3.4"
+	svcPortName := proxy.ServicePortName{
+		NamespacedName: makeNSN("ns1", "svc1"),
+		Port:           "p80",
+	}
+
+	makeServiceMap(fp,
+		makeTestService(svcPortName.Namespace, svcPortName.Name, func(svc *v1.Service) {
+			svc.Spec.Type = "LoadBalancer"
+			svc.Spec.ClusterIP = svcIP
+			svc.Spec.Ports = []v1.ServicePort{{
+				Name:     svcPortName.Port,
+				Port:     int32(svcPort),
+				Protocol: v1.ProtocolTCP,
+				NodePort: int32(svcNodePort),
+			}}
+			svc.Status.LoadBalancer.Ingress = []v1.LoadBalancerIngress{{
+				IP:        svcLBIP,
+				RouteType: v1.LoadBalancerRouteTypeProxy,
+			}}
+		}),
+	)
+
+	epIP := "10.180.0.1"
+	makeEndpointsMap(fp,
+		makeTestEndpoints(svcPortName.Namespace, svcPortName.Name, func(ept *v1.Endpoints) {
+			ept.Subsets = []v1.EndpointSubset{{
+				Addresses: []v1.EndpointAddress{{
+					IP: epIP,
+				}},
+				Ports: []v1.EndpointPort{{
+					Name: svcPortName.Port,
+					Port: int32(svcPort),
+				}},
+			}}
+		}),
+	)
+
+	fp.syncProxyRules()
+
+	services, err := fp.ipvs.GetVirtualServers()
+	if err != nil {
+		t.Errorf("Failed to get ipvs services, err: %v", err)
+	}
+	if len(services) != 1 {
```

Comment:
not familiar with ipvs, but curious about this behaviour, in this test, if `LoadBalancerRouteTypeProxy == VIP` , len(services) should be 2?

Reply:
In the default and current behaviour, another virtual server is created for the LB IP iirc

Reply:
can we use a table test then?

```go
testCases := []struct {
		svcIP string
                svcLBIP string
                routeType v1.LoadBalancerRouteType
		expectedServices    int
	}{
		// case 0
		{
			svcIP: "10.20.30.41",
			svcLBIP:    "1.2.3.4",
			routeType:    v1.LoadBalancerRouteTypeProxy,
                         expectedServices: 1,
		},
		// case 1
		{
			svcIP: "10.20.30.41",
			svcLBIP:    "1.2.3.4",
			routeType:    v1.LoadBalancerRouteTypeVIP,
                         expectedServices: 2,
		},
		// case 2
		{
			svcIP: "fd00:1234::1",
			svcLBIP:    "2001:db8::1",
			routeType:    v1.LoadBalancerRouteTypeProxy,
                         expectedServices: 1,
		},
		// case 3
		{
			svcIP: "fd00:1234::1",
			svcLBIP:    "2001:db8::1",
			routeType:    v1.LoadBalancerRouteTypeVIP,
                         expectedServices: 2,
		},
		// case 4
		{
			svcIP: "fd00:1234::1",
			svcLBIP:    "2001:db8::1",
			routeType:    "",
                         expectedServices: 2,
		},
```